qiskit/transpiler/passes/utils/check_map.py

```diff
@@ -46,20 +65,18 @@ def run(self, dag):
 
         self.property_set["is_swap_mapped"] = True
 
-        if self.coupling_map is None or len(self.coupling_map.graph) == 0:
+        if not self.qargs:
             return
```

Comment:
THis is kind of pre-existing, but silently passing through on `len(self.coupling_map.graph) == 0` seems wrong to me; the given coupling map is saying "there are no allowed 2q operations", so that should be a failure, right?

Reply:
Oh actually `len(self.coupling_map.graph) == 0`  is saying there are no qubits on the device (`len()` on a rustworkx graph is the number of nodes), I'm not actually sure what this means in this case. I think there is/was probably some weird edge case where this got triggered and we added the condition, but I don't really remember/know.